### DISCUSSION TITLE
Button as Link doesn't work as intended

### DISCUSSION THREAD

**Discussion by dholliday3** [twilio-labs/paste#2550] [General] [answered]
I want to use a button as a link specifically because the Anchor tag is underlined which I don't want. 
In the docs there's an example of exactly what I want with a [link style button](https://paste.twilio.design/components/button/#link-style-button) but when I try to add a link it doesn't work. 

**Here's what I'm trying to use:** 
![Screen Shot 2022-07-21 at 4 12 56 PM](https://user-images.githubusercontent.com/15958912/180330209-34be83f4-a39f-47d2-a01d-38e925ebfa45.png)

**When I add an href I get the expected warning that I need to use "as='a'":** 
![Screen Shot 2022-07-21 at 4 09 52 PM](https://user-images.githubusercontent.com/15958912/180330314-d9b2d17c-85af-4c3b-9cd3-9350aead2d57.png)

**But when I do what the warning says and add "as='a'" it gives me this warning and basically renders the button link unusable.**
![Screen Shot 2022-07-21 at 4 10 04 PM](https://user-images.githubusercontent.com/15958912/180330385-bfce016c-f50d-45dd-9db4-3fa59594bfb9.png)

Again, the trouble here is that I don't want the button to be underlined but it seems like the main workaround is to pass an onClick and explicitly handle navigation. 
Would love to be able to use the link style button as intended. 

Thanks for the help!

**Comment by shleewhite** [comment, marked as answer]
Hi @dholliday3

Can you explain your use case for an anchor with no underline?

In general, buttons perform actions on the page the user is currently on and anchors take the user to another page. For consistency and clarity for users, Paste has a rule that all anchors have an underline and buttons do not have an underline (we have documentation about this in [the Button vs. Anchor pattern](https://paste.twilio.design/patterns/button-vs-anchor/#accessibility)).

The link style button doesn’t have an underline because even though it looks _similar_ to an anchor, it is a button. We throw an error when you try to make a link style button an anchor because generally, it is preferable to use the actual Anchor component. The only time we only allow [a Button with `as=“a”`](https://paste.twilio.design/components/button/#buttons-with-link-functionality) if it is the primary or secondary variant.

If you need something that takes the user to another page, I would strongly suggest using the Anchor component.

**Reply by dholliday3** [reply]
The use case is to click on a title of a card in a grid that will go to another page. This is kind of similar to the buttons used in the menu for the Paste website. 

![image](https://user-images.githubusercontent.com/15958912/180879098-1a8c8b7c-c165-4497-8973-e5a8e476a4f4.png)
 
I definitely understand the difference between the two components (anchor vs link-style button) but I think it would make sense to allow the use case where we include an href in a button.

**Reply by shleewhite** [reply]
@dholliday3 Could you share a screen shot of the design? I think that would help us understand the situation better.

**Reply by dholliday3** [reply]
Sure! This is what it looks like right now. 

![Screen Shot 2022-07-25 at 5 40 10 PM](https://user-images.githubusercontent.com/15958912/180898260-28969419-fb51-41ec-b79c-ce9871e76ee4.png)

**Reply by shleewhite** [reply]
@dholliday3 Thanks for sharing! We discussed and this UI seems like a valid use case for an anchor with no underline because there are no other links in the card and these kind of marketplace UIs are very common. 

Instead of using the Button or Anchor components, you can use Box to make your own custom anchor. Here is an example of how you could build it: https://codesandbox.io/s/custom-anchor-fgfkgr

**Reply by dholliday3** [reply]
Awesome - thanks! 

Is there a plan to support this use-case natively? In the meantime, I could go the custom route but I'd love to see this come out of the box with either button or anchor. 

Is this something that I can make a contribution to?

**Reply by shleewhite** [reply]
Unfortunately because this is an edge case, there are no plans to add support for this in the Button or Anchor component.
